### DISCUSSION TITLE
Need to get rid of cursor dot on area graph while composed graph is has scatter on top

### DISCUSSION THREAD

**Discussion by ToeJammm** [recharts/recharts#4684] [Q&A] [answered]
The cursor dot that aligns with the cursor while I am hovering above my area graph works great until I use a button to add a scatter plot on top of the area graph. It lags quite far behind the cursor for some reason, so I figured I would just get rid of it, but I cannot figure out how to disable it. If someone knows how to get it to behave correctly or get rid of it please let me know. I have attached an image of my issue. 
![rechartsIssue](https://github.com/recharts/recharts/assets/114703597/0aedec81-a530-4cd2-9d30-3b16c90b9a84)

**Comment by ckifer** [comment]
Can you create a reproduction of the issue? I'd imagine this is with `ComposedChart`?
https://codesandbox.io/p/sandbox/composed-chart-with-best-fit-forked-rrkvz2

**Reply by ToeJammm** [reply]
I am currently trying to replicate my situation in the sandbox but it  appears there is no dot that follows my cursor on the graph when I just simply have an area graph or an area + scatter. Where do you think this dot is coming from? my tooltip or the graph itself. I remember it being there since the beginning of the area data being rendered.

**Reply by ckifer** [reply]
🤔. Weird. Do you have anything custom going on in your code? With Tooltip, Cursor, or Dots?

AFAIK recharts doesn't have the "follow" functionality out of the box

**Reply by ToeJammm** [reply]
I have a custom tooltip that I added into the sandbox but that didn't cause any dots

**Reply by ToeJammm** [reply]
![Screenshot 2024-06-20 161029](https://github.com/recharts/recharts/assets/114703597/b3c411ee-6936-4e5b-8b2b-19f42cbab95b)
It usually follows the cursor like this. I liked it at first so I never questioned it.

**Reply by ckifer** [reply, marked as answer]
Ah right, so it follows the "cursor component" (the line up and down, or the active index) but it doesn't follow the mouse exactly right? It jumps around?

That can be turned off with `activeDot={false}`

Also you'll need to share your sandbox link since it's a fork of mine

**Reply by ToeJammm** [reply]
other than my custom tooltip, I do have some variables that keep track of where my cursor is for certain functions like zooming in or choosing certain frequencies

**Reply by ToeJammm** [reply]
https://codesandbox.io/p/sandbox/composed-chart-with-best-fit-forked-hd9ns7?file=%2Fsrc%2FApp.tsx%3A66%2C22

**Reply by ckifer** [reply]
Can you make your sandbox public 
![image](https://github.com/recharts/recharts/assets/25180830/4890cfb7-8161-48e5-afa0-3e3863f5ea75)

**Reply by ToeJammm** [reply]
that worked! thankyou (:

**Reply by ckifer** [reply]
Well, I'm not sure why it doesn't "just work", but if you're keeping track of coordinates and providing them back to recharts at some point it may cause conflict with the default behavior. Without seeing all of your code its hard to tell otherwise

**Reply by ToeJammm** [reply]
Sorry, I have one more question here. I would still like the cursor to follow the area graph while my scatter graph is being rendered. Currently whenever I toggle the scatter graph on, the legend only acknowledges the points of the scatter and not the area graph. I've tried putting area above and behind the scatter to see if it would make a difference but it did not. How do I get the cursor to only focus on the points of the area graph?

**Reply by ckifer** [reply]
what do you mean by the legend? The Legend component? That doesn't seem to relate to the cursor or where your mouse is at

**Reply by ToeJammm** [reply]
yes the legend, but It seems to only allow me to select vertices of the scatter graph when the scatter is being rendered with the area graph. The legend and the possible values the cursor can click seem to be tied together. I'll attach a short clip.

https://github.com/recharts/recharts/assets/114703597/fd1d887e-38ef-448a-b238-b343bb5b18b5

**Reply by ckifer** [reply]
You don't have a legend, or at least you don't have a `Legend` from recharts unless I'm seeing things wrong or your legend is very customized.

It looks like you are toggling things on and off with some buttons? And then moving focus of your scatter plots with more buttons?
The cursor is snapping to the scatter dots, but it looks like you can still get datapoints from the Area if you hover closer to them.

Would replacing your scatter with `ReferenceDot`s be acceptable? They wouldn't have the same effect on your cursor snapping, but they also wouldn't render in the Tooltip

**Reply by ToeJammm** [reply]
I think that would be acceptable I really don't care about the snapping. How would I do that since it would be a much different process than just giving the scatter plot a data key.

**Reply by ckifer** [reply]
You would need to loop over your datapoints which contain the scatter keys
```
      {data
        .filter((d) => d.blue > 0)
        .map((d) => (
          <ReferenceDot
            fill="blue"
            r={5}
            x={d?.index}
            y={d?.blue}
            ifOverflow="extendDomain"
          />
        ))}
```

**Reply by ckifer** [reply]
https://codesandbox.io/p/sandbox/composed-chart-with-best-fit-forked-rrkvz2?file=%2Fsrc%2FApp.tsx%3A57%2C1-67%2C12

**Reply by ToeJammm** [reply]
ok that makes sense to me. Again thanks so much for your help. Front end support like this seems rare to me.

**Reply by ckifer** [reply]
No worries! Hope you can accomplish what you need to and feel free to create issues if you can't
